src/components/MagicCodeInput.js

```diff
@@ -95,6 +95,7 @@ function MagicCodeInput(props) {
     const [input, setInput] = useState('');
     const [focusedIndex, setFocusedIndex] = useState(0);
     const [editIndex, setEditIndex] = useState(0);
+    const [firstSubmit, setFirstSubmit] = useState(true);
```

Comment:
Would you have something against naming this `wasSubmitted` (and flipping false/true)? It would feel more natural to me

Reply:
Yes, that's my suggestion. Names would be `wasSubmitted` / `setWasSubmitted`.

Reply:
> and flipping false/true

Please note that "was (already) submitted?" is a negation of "(is this a) first submit?". We need to flip the boolean logic.